### PR TITLE
Add -time-file to output the time data to a file

### PR DESCRIPTION
**Pull request by SkySkimmer**
instead of having to redirect stdout

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by Alizter** [comment]
What about calling it `-time-file` or `-time-with-file` instead so that it is a bit more descriptive.

**Review by Alizter — APPROVED** [review]
Otherwise looks good.

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Review by ejgallego — COMMENTED** [review]
LGTM, it would be great if when outputting to a file we would produce a more structured format instead of the current one.

**Comment by ejgallego** [comment]
> The current format seems relatively structured though?

Maybe, but it is even documented ?

cc #11542

**Review by ejgallego — APPROVED** [review]

**Comment by Alizter** [comment]
@JasonGross If you get a chance, could you also have a look over in case anything stands out that isn't quite right.

File: doc/changelog/09-cli-tools/17430-timef.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Added:**
+  option `-timef`, like `time` but outputting to a file
```
Comment (JasonGross) [inline doc/changelog/09-cli-tools/17430-timef.rst:2]: This should have been updated when the flag was renamed

**Review by JasonGross — COMMENTED** [review]
Nothing stands out to me other than the outdated name of the flag on the changelog.  And I'm glad this no longer requires redirect hacks!  (Maybe there'll now be a way to have the dune build of Coq itself generate timing files when the right flags are passed to dune?  cc @ejgallego ?)

**Comment by Alizter** [comment]
@JasonGross Yes, it should be pretty simple to add some extra stuff to the rule generation passing `"-time-file"; vfile ^ ".timings"` or something.

**Comment by ejgallego** [comment]
@JasonGross the dune build of coq itself already passes the flag

**Pull request merged**